### PR TITLE
wireguard-tools: clean up, wrap w/ wireguard-go on Darwin

### PR DESCRIPTION
**Pull request by lukateras**
@kirelagin @zx2c4 

Cleans up derivation (consolidates some things), also note that systemd unit is now built unconditionally (it is just a text file anyway). On Darwin, `wg` and `wg-quick` are wrapped together with `wireguard-go`.

### CODE REVIEW

File: pkgs/tools/networking/wireguard-tools/default.nix [hunk 1]
```diff
@@ -1,46 +1,43 @@
...
       --replace /usr/bin $out/bin
+  '' + optionalString stdenv.isDarwin ''
+    for f in $out/bin/*; do
+      wrapProgram $f --prefix PATH : ${wireguard-go}/bin
+    done
   '';
 
   meta = with stdenv.lib; {
-    homepage     = https://www.wireguard.com/;
-    downloadPage = https://git.zx2c4.com/WireGuard/refs/;
```
[the remark this answers is not in the document]
Comment (yegortimoshenko) [reply pkgs/tools/networking/wireguard-tools/default.nix]: Didn't mean to remove. You noticed faster than I could amend :-)

**Comment by zx2c4** [comment]
:+1:

File: pkgs/tools/networking/wireguard-tools/default.nix [hunk 2]
```diff
@@ -1,46 +1,44 @@
...
 
   makeFlags = [
-    "WITH_BASHCOMPLETION=yes"
-    "WITH_WGQUICK=yes"
-    "WITH_SYSTEMDUNITS=${if useSystemd then "yes" else "no"}"
     "DESTDIR=$(out)"
     "PREFIX=/"
-    "-C" "tools"
+    "WITH_BASHCOMPLETION=yes"
+    "WITH_SYSTEMDUNITS=yes"
```
Comment (zx2c4) [inline pkgs/tools/networking/wireguard-tools/default.nix]: Why not keep the useSystemd mark in here? It's indeed really useless on macOS.

Comment (yegortimoshenko) [reply pkgs/tools/networking/wireguard-tools/default.nix]: Smaller variability helps package maintenance. In this case, this required conditional patch in `postFixup` phase. Given that it's just a text file, it seems to easier to copy and patch anyway than to decide based on platform.

**Pull request merged**